web-src/src/components/GenerateButton.js

```diff
@@ -74,12 +74,22 @@ export function GenerateButton() {
     sampleRUM('genai:prompt:generate', { source: 'GenerateButton#handleGenerate' });
     setGenerationInProgress(true);
     setIsOpenPromptEditor(false);
+    const startTime = Date.now();
     generateResults()
       .catch((error) => {
         ToastQueue.negative(error.message, { timeout: 2000 });
+
+        // capture network errors
+        if (error.status > 300) {
```

Comment:
I'd separate [Client error responses](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses) (400 – 499) and [Server error responses](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#server_error_responses) (500 – 599).

Reply:
+1